projection/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -597,4 +606,25 @@ private[projection] class R2dbcOffsetStore(
     Future.successful(Done) // FIXME not implemented yet
   }
 
+  private def createRecordWithOffset[Envelope](envelope: Envelope): Option[RecordWithOffset] = {
+    envelope match {
+      case eventEnvelope: EventEnvelope[_] if eventEnvelope.offset.isInstanceOf[TimestampOffset] =>
+        val timestampOffset = eventEnvelope.offset.asInstanceOf[TimestampOffset]
+        Some(
+          RecordWithOffset(
+            Record(eventEnvelope.persistenceId, eventEnvelope.sequenceNr, timestampOffset.timestamp),
+            timestampOffset,
+            strictSeqNr = true))
+      case change: UpdatedDurableState[_] if change.offset.isInstanceOf[TimestampOffset] =>
+        val timestampOffset = change.offset.asInstanceOf[TimestampOffset]
+        Some(
+          RecordWithOffset(
+            Record(change.persistenceId, change.revision, timestampOffset.timestamp),
+            timestampOffset,
+            strictSeqNr = false))
+      // FIXME case DeletedDurableState when that is added
+      case _ => None
```

Comment:
Should that be a match error/exception rather and this method return `RecordWithOffset` directly? Because it is an error if we forget to implement it for DeletedDurableState when that is added, or if we get some non TimestampOffset envelope here.

Reply:
yes, I'll change that